src/ansys/aedt/core/modeler/cad/object_3d.py

```diff
@@ -2030,24 +2030,30 @@ def chamfer(self, vertices=None, edges=None, left_distance=1, right_distance=Non
         vArg2 = ["NAME:ChamferParameters"]
         vArg2.append("Edges:="), vArg2.append(edge_id_list)
         vArg2.append("Vertices:="), vArg2.append(vertex_id_list)
-        vArg2.append("LeftDistance:="), vArg2.append(self._primitives._arg_with_dim(left_distance))
-        if not right_distance:
+        if right_distance is None:
             right_distance = left_distance
         if chamfer_type == 0:
+            if left_distance != right_distance:
+                self.logger.error("Do not set right distance or ensure that left distance equals right distance.")
+            vArg2.append("LeftDistance:="), vArg2.append(self._primitives._arg_with_dim(left_distance))
             vArg2.append("RightDistance:="), vArg2.append(self._primitives._arg_with_dim(right_distance))
             vArg2.append("ChamferType:="), vArg2.append("Symmetric")
         elif chamfer_type == 1:
+            vArg2.append("LeftDistance:="), vArg2.append(self._primitives._arg_with_dim(left_distance))
             vArg2.append("RightDistance:="), vArg2.append(self._primitives._arg_with_dim(right_distance))
             vArg2.append("ChamferType:="), vArg2.append("Left Distance-Right Distance")
         elif chamfer_type == 2:
-            vArg2.append("Angle:="), vArg2.append(str(angle) + "deg")
-            vArg2.append("ChamferType:="), vArg2.append("Left Distance-Right Distance")
+            vArg2.append("LeftDistance:="), vArg2.append(self._primitives._arg_with_dim(left_distance))
+            # NOTE: Seems like there is a bug in the API as Angle can't be used
+            vArg2.append("RightDistance:="), vArg2.append(f"{angle}deg")
```

Comment:
Just to be sure, is it expected to provide an `angle` as the `RightDistance`?

Reply:
There is a bug and this is the work around. You can't pass angle through `"Angle:="` and need to use the distance property instead.